infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/WaitSnapshotApplyState.java

```diff
@@ -102,6 +109,11 @@ public LogReplicationState processEvent(LogReplicationEvent event) throws Illega
                  Snapshot Sync with ID = 1 could be completed in between (1 and 2) but show up in the queue
                  as 4, attempting to process a completion event for the incorrect snapshot sync.
                  */
+
```

Comment:
Let's add a comment on why the flag is changed and why this check is done here..